core/data/dictionary/dictionary.go

```diff
@@ -23,44 +23,68 @@ import (
 
 // I is the interface for a dictionary.
 type I interface {
+	// Get returns the value of the entry with the given key.
 	Get(key interface{}) (val interface{})
-	Set(key, val interface{})
-	Lookup(key interface{}) (val interface{}, ok bool)
+	// Add inserts the key-value pair, replacing any existing entry with the
+	// same key.
+	Add(key, val interface{})
+	// Lookup searches for the value of the entry with the given key.
+	Lookup(key interface{}) (val interface{}, found bool)
+	// Contains returns true if the dictionary contains an entry with the given
+	// key.
 	Contains(key interface{}) bool
-	Delete(key interface{})
-	Clear()
```

Comment:
Should we keep `Clear`? I know go doesn't have a way to clear a map, but I don't think we should impose that on other implementations. The overhead of delete-one-at-atime to delete-all-at-once, especially with the keys sorting, seems like it can get quite large.

Reply:
I'm trying to keep the interface surface small.
There are actually no used of `Clear` in our codebase (with exception to the test). If we find in the future that it is a bottleneck, I'd probably opt for creating a `Clearer` interface that `Clear` uses as an optional fast path.